File: GXpert/GXpert.Web/Modules/Content/ContentCourse/ContentCourseColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Content.Columns;

[ColumnsScript("Content.ContentCourse")]
[BasedOnRow(typeof(ContentCourseRow), CheckNames = true)]
public class ContentCourseColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ContentTitle { get; set; }
    public string CourseTitle { get; set; }
    public string ClassTitle { get; set; }
    public string SemesterTitle { get; set; }
    public string SubjectTitle { get; set; }
    public string TopicTitle { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentCourseColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContentCourseRow } from "./ContentCourseRow";

export interface ContentCourseColumns {
    Id: Column<ContentCourseRow>;
    ContentTitle: Column<ContentCourseRow>;
    CourseTitle: Column<ContentCourseRow>;
    ClassTitle: Column<ContentCourseRow>;
    SemesterTitle: Column<ContentCourseRow>;
    SubjectTitle: Column<ContentCourseRow>;
    TopicTitle: Column<ContentCourseRow>;
}

export class ContentCourseColumns extends ColumnsBase<ContentCourseRow> {
    static readonly columnsKey = 'Content.ContentCourse';
    static readonly Fields = fieldsProxy<ContentCourseColumns>();
}